File: backend/models/exercise_request.py
from enum import Enum
from sqlalchemy import Column, Integer, String, Enum as SQLEnum
from backend.database import Base
from backend.models.exercise import MuscleGroupEnum


class RequestStatusEnum(str, Enum):
    PENDING = "pending"
    APPROVED = "approved"
    DENIED = "denied"


class ExerciseRequest(Base):
    __tablename__ = "exercise_requests"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, nullable=False, index=True)
    suggested_name = Column(String(100), nullable=False)
    muscle_group = Column(SQLEnum(MuscleGroupEnum), nullable=False)
    status = Column(
        SQLEnum(RequestStatusEnum),
        nullable=False,
        default=RequestStatusEnum.PENDING,
        index=True,
    )
    

File: frontend/src/schemas/exercise-request.ts
import { z } from 'zod'
import { MuscleGroupSchema } from './exercise'

export const RequestStatusSchema = z.enum(['pending', 'approved', 'denied'])
export type RequestStatus = z.infer<typeof RequestStatusSchema>

export const ExerciseRequestSchema = z.object({
  id: z.number(),
  user_id: z.number(),
  suggested_name: z.string(),
  muscle_group: MuscleGroupSchema,
  is_cardio: z.boolean(),
  status: RequestStatusSchema,
})
export type ExerciseRequest = z.infer<typeof ExerciseRequestSchema>

export const ExerciseRequestCreateSchema = z.object({
  suggested_name: z.string(),
  muscle_group: MuscleGroupSchema,
  is_cardio: z.boolean(),
})
export type ExerciseRequestCreate = z.infer<typeof ExerciseRequestCreateSchema>
